examples/chatpromptbuilder.py

```diff
@@ -0,0 +1,37 @@
+from haystack.components.builders import ChatPromptBuilder
+from haystack.components.generators.chat.openai import OpenAIChatGenerator
+from haystack.dataclasses.chat_message import ImageContent
+
+### Use multimodal inputs with ChatPromptBuilder
+
+template = """
+{% message role="system" %}
+    You are a helpful and joking assistant.
+{% endmessage %}
+
+{% message role="user" %}
+    What's the difference between the two images?
+    {% for image in images %}
+        {{ image | for_template }}
+    {% endfor %}
```

Comment:
How does this work exactly? As in how does this properly get translated into a textcontent and a list of imagecontents internally? Like if I did this as a user
```python
    What's the difference between the two images?
    {% for image in images %}
        I'm adding text here too {{ image | for_template }}
    {% endfor %}
```
what would happen?